src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/FilesystemController.kt

```diff
@@ -175,4 +190,24 @@ class FilesystemController(
     fun notFoundException(ex: NotFoundException): ModelAndView {
         return ModelAndView("404", "path", ex.path)
     }
+
+    @Secured(Permission.MANAGE_FILES_STR)
+    @DeleteMapping("/delete/**")
+    fun deleteResource(request: HttpServletRequest): ModelAndView {
+        val path = request.servletPath.removePrefix("/delete/")
+        val entity = repoService.getEntity(path) ?: throw NotFoundException(path)
+        val mountPathElements = if (path.endsWith("/")) path.dropLast(1).split("/") else path.split("/")
+        val mountPath = mountPathElements.joinToString("/")
+        if (entity is RepositoryDirectory) {
+            repoService.delete(entity.relativePath, true)
```

Comment:
are you sure this should be relativePath not path?

Reply:
I am :D

`@Transactional
    fun delete(path: String, recursive: Boolean = false) {
        val absolutePath = getAbsolutePath(path)
        val entity = getEntity(path)`